pkg/provider/provider.go

```diff
@@ -622,20 +629,29 @@ func (p *Provider) buildHTTPServiceFromService(t *topology.Topology, svc *topolo
 	}
 }
 
-func (p *Provider) buildHTTPServiceFromTrafficTarget(t *topology.Topology, tt *topology.ServiceTrafficTarget, scheme string, port int32) *dynamic.Service {
-	servers := make([]dynamic.Server, len(tt.Destination.Pods))
+func (p *Provider) buildHTTPServiceFromTrafficTarget(t *topology.Topology, tt *topology.ServiceTrafficTarget, scheme string, svcPort corev1.ServicePort) *dynamic.Service {
+	var servers []dynamic.Server
 
-	for i, podKey := range tt.Destination.Pods {
+	for _, podKey := range tt.Destination.Pods {
 		pod, ok := t.Pods[podKey]
 		if !ok {
 			p.logger.Errorf("Unable to find Pod %q for HTTP service from Traffic Target %s@%s", podKey, topology.Key{Name: tt.Name, Namespace: tt.Namespace})
```

Comment:
Not related to your changes but we have an issue with this log message. It should be:
```suggestion
			p.logger.Errorf("Unable to find Pod %q for HTTP service from Traffic Target %q", podKey, topology.ServiceTrafficTargetKey{
				Service:       tt.Service,
				TrafficTarget: topology.Key{Name: tt.Name, Namespace: tt.Namespace},
			})
```
If you think it's the proper time to fix this, I have seen this mistake couple of time in this file.